src/base/memory_monitor_sr.hpp

```diff
@@ -77,13 +106,111 @@ namespace cubmem
       // and some C++ allocations occurred at header files.
       const int m_magic_number;
   };
+
+  extern memory_monitor *mmon_Gl;
+
+  inline int memory_monitor::get_target_pos ()
+  {
+    return m_target_pos;
+  }
+
+  inline void memory_monitor::make_stat_name (char *buf, const char *file, const int line)
+  {
+    sprintf (buf, "%s:%d", file + m_target_pos, line);
+  }
+
+  inline char *memory_monitor::get_metainfo_pos (char *ptr, size_t size)
+  {
+    return ptr + size - MMON_METAINFO_SIZE;
+  }
+
+  inline void memory_monitor::add_stat (char *ptr, const size_t size, const char *file, const int line)
```

Comment:
I'm doubtful whether it's appropriate to write this function as an inline function due to the size of the code. Are there many parts calling `add_stat()`? If there are many parts calling `add_stat/sub_stat`, it seems better to write it as a regular function instead of an inline function

Reply:
The method is only called by API (mmon_add/sub_stat()) and the APIs are called allocation wrapping functions.

Reply:
OK, then we can leave this as it is.